src/app/modules/form/form.service.ts

```diff
@@ -120,3 +123,25 @@ export const isFormPublic = (
       return assertUnreachable(form.status)
   }
 }
+
+/**
+ * Returns public view for given form, removing all private details.
+ * @param form the form to return public view for
+ * @returns ok(public form view)
+ * @returns err(Database*Errors) if any database error occurs.
+ */
+export const getFormPublicView = (
+  form: IFormSchema,
+): ResultAsync<PublicForm, ReturnType<typeof transformMongoError>> => {
+  return ResultAsync.fromPromise(form.getPublicView(), (error) => {
+    logger.error({
+      message: 'Error retrieving form public view',
+      meta: {
+        action: 'getFormPublicView',
+        formId: form._id,
+      },
+      error,
+    })
+    return transformMongoError(error)
+  })
+}
```

Comment:
This service method is only needed because `form.getPublicView()` implementation is async, which in turn is because it contains the implementation for populating the `admin` key. Could `form.getPublicView()` be implemented such that it is agnostic to whether the form object is populated or not?

In fact one could even implement `agency.getPublicView()` and `user.getPublicView()`

Reply:
so getPublicView just returns the subset of the form document without populating (if its not populated)?

so for non-populated form `nonPopulatedForm.getPublicView()` will return an unpopulated agency id, yes?